src/qtcwin.c

```diff
@@ -0,0 +1,1814 @@
+			posidx = 5;
+			break;
+	    }
+	    if (pressed == '?') {
+		//if (qtcreclist.qtclines[qtcrow].status == 2) {
+		    //qtcreclist.confirmed--;
+		//}
+		qtcreclist.qtclines[qtcrow].status = 1;	// set incomplete the qtc status
+		show_status(qtcrow);
+	    }
+	    if ( ( ( (stridx == 0 || stridx == 2) && (isdigit(pressed) || pressed == '?') ) || stridx == 1) && strlen(fieldval) < pos[posidx][2]) {
+		shift_right(fieldval);
+		fieldval[strlen(fieldval)-curpos] = pressed;
+		fieldval[strlen(fieldval)+1] = '\0';
+		switch(stridx) {
+		    case 0: strcpy(qtcreclist.qtclines[qtcrow].time, fieldval);
+			    break;
+		    case 1: strcpy(qtcreclist.qtclines[qtcrow].callsign, fieldval);
+			    break;
+		    case 2: strcpy(qtcreclist.qtclines[qtcrow].serial, fieldval);
+			    break;
+		}
+		if (curpos > 0) {
+		    curpos--;
+		}
+		showfield(activefield);
+		if (stridx == 0 && strlen(fieldval) == pos[posidx][2] && activefield <= 32) {	// auto TAB if curr field is QTC time, and len is 4
+		    curpos = 0;
+		    activefield++;
+		    showfield(activefield-1);
+		    showfield(activefield);
+		}
+	    }
+
+	}
+	return 0;
+}
+
+int delete_from_field(int dir) {
+
+	char fieldval[16];
+	int fi, winrow, qtcrow, stridx;
+
+	if (activefield == 0) {
+	    if (strlen(qtccallsign) > 0) {
+		sprintf(fieldval, "%s", qtccallsign);
+		shift_left(fieldval, dir);
+		strncpy(qtccallsign, fieldval, strlen(fieldval));
+		qtccallsign[strlen(fieldval)] = '\0';
+		recalc_qtclist();
+		showfield(0);
+	    }
+	}
+	else if (activefield == 1) {
+	    sprintf(fieldval, "%d", qtcreclist.serial);
+	    if(strlen(fieldval) > 0) {
+	      //sprintf(fieldval, "%d", qtcreclist.serial/10);
+	      shift_left(fieldval, dir);
+	      qtcreclist.serial = atoi(fieldval);
+	      showfield(1);
+	    }
+	}
+	else if (activefield == 2) {
+	    sprintf(fieldval, "%d", *qtccount);
+	    if(strlen(fieldval) > 0) {
+	      //sprintf(fieldval, "%d", qtcreclist.count/10);
+	      shift_left(fieldval, dir);
+	      *qtccount = atoi(fieldval);
+	      showfield(2);
+	    }
+	}
+	else {
+	    fi = activefield-3;
+	    winrow = (fi/3)+3;
+	    qtcrow = winrow-3;
+	    stridx = fi%3;
+	    switch(stridx) {
+		case 0:	sprintf(fieldval, "%s", qtcreclist.qtclines[qtcrow].time);
+			break;
+		case 1:	sprintf(fieldval, "%s", qtcreclist.qtclines[qtcrow].callsign);
+			break;
+		case 2:	sprintf(fieldval, "%s", qtcreclist.qtclines[qtcrow].serial);
+			break;
+	    }
+
+	    if (strlen(fieldval) > 0) {
+		//fieldval[strlen(fieldval)-2] = '\0';
+		shift_left(fieldval, dir);
+		//strncpy(newfieldval, fieldval, strlen(fieldval)-2);
+		switch(stridx) {
+		    case 0: strcpy(qtcreclist.qtclines[qtcrow].time, fieldval);
+			    break;
+		    case 1: strcpy(qtcreclist.qtclines[qtcrow].callsign, fieldval);
+			    break;
+		    case 2: strcpy(qtcreclist.qtclines[qtcrow].serial, fieldval);
+			    break;
+		}
+		showfield(activefield);
+	    }
+
+	}
+	return 0;
+}
+
+int shift_right(char * fieldval) {
```

Comment:
Adding a second \0 to terminate, but then it returns, the next code could insert a charater to the current position. What do ou propose to handle this inssert function?
It could be better to make a function with arguments field and character, eg "insert_char(field, c)"? What do you think?